Python3.12.4 breaks two tests

Hi, thanks for developing an awesome library.
I maintain the Python ecosystem on NixOS and noticed that two tests (`test_start_connection_cannot_increase_recv_buffer` and `test_start_connection_can_only_increase_buffer_size_to_262144`) failed due to changes in Python 3.12.3 -> 3.12.4.
Here is the log and the [issue submitted to CPython](https://github.com/python/cpython/issues/120732).

This bug can also be found in the [GHA](https://github.com/esphome/aioesphomeapi/actions/runs/9580529837/job/26415337755).

```
aioesphomeapi> =================================== FAILURES ===================================
aioesphomeapi> ______________ test_start_connection_cannot_increase_recv_buffer _______________
aioesphomeapi> 
aioesphomeapi> conn = <tests.conftest.PatchableAPIConnection object at 0x7ffff43b23f0>
aioesphomeapi> resolve_host = <AsyncMock name='async_resolve_host' id='140737289717424'>
aioesphomeapi> aiohappyeyeballs_start_connection = <AsyncMock name='start_connection' id='140737293553008'>
aioesphomeapi> caplog = <_pytest.logging.LogCaptureFixture object at 0x7ffff4791040>
aioesphomeapi> 
aioesphomeapi>     @pytest.mark.asyncio
aioesphomeapi>     async def test_start_connection_cannot_increase_recv_buffer(
aioesphomeapi>         conn: APIConnection,
aioesphomeapi>         resolve_host,
aioesphomeapi>         aiohappyeyeballs_start_connection: MagicMock,
aioesphomeapi>         caplog: pytest.LogCaptureFixture,
aioesphomeapi>     ):
aioesphomeapi>         """Test failing to increase the recv buffer."""
aioesphomeapi>         loop = asyncio.get_event_loop()
aioesphomeapi>         transport = MagicMock()
aioesphomeapi>         connected = asyncio.Event()
aioesphomeapi>         tried_sizes = []
aioesphomeapi>     
aioesphomeapi>         def _setsockopt(*args, **kwargs):
aioesphomeapi>             if args[0] == socket.SOL_SOCKET and args[1] == socket.SO_RCVBUF:
aioesphomeapi>                 size = args[2]
aioesphomeapi>                 tried_sizes.append(size)
aioesphomeapi>                 raise OSError("Socket error")
aioesphomeapi>     
aioesphomeapi> >       mock_socket: socket.socket = create_autospec(
aioesphomeapi>             socket.socket, spec_set=True, instance=True, name="bad_buffer_socket"
aioesphomeapi>         )
aioesphomeapi> 
aioesphomeapi> tests/test_connection.py:264: 
aioesphomeapi> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
aioesphomeapi> /nix/store/2cvwgfc1cd1c2ii4b08r99m4wrdqqqdj-python3-3.12.4/lib/python3.12/unittest/mock.py:2823: in create_autospec
aioesphomeapi>     mock.configure_mock(**kwargs)
aioesphomeapi> /nix/store/2cvwgfc1cd1c2ii4b08r99m4wrdqqqdj-python3-3.12.4/lib/python3.12/unittest/mock.py:650: in configure_mock
aioesphomeapi>     setattr(obj, final, val)
aioesphomeapi> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
aioesphomeapi> 
aioesphomeapi> self = <NonCallableMagicMock name='bad_buffer_socket' spec_set='socket' id='140737288272976'>
aioesphomeapi> name = 'name', value = 'bad_buffer_socket'
aioesphomeapi> 
aioesphomeapi>     def __setattr__(self, name, value):
aioesphomeapi>         if name in _allowed_names:
aioesphomeapi>             # property setters go through here
aioesphomeapi>             return object.__setattr__(self, name, value)
aioesphomeapi>         elif (self._spec_set and self._mock_methods is not None and
aioesphomeapi>             name not in self._mock_methods and
aioesphomeapi>             name not in self.__dict__):
aioesphomeapi> >           raise AttributeError("Mock object has no attribute '%s'" % name)
aioesphomeapi> E           AttributeError: Mock object has no attribute 'name'
aioesphomeapi> 
aioesphomeapi> /nix/store/2cvwgfc1cd1c2ii4b08r99m4wrdqqqdj-python3-3.12.4/lib/python3.12/unittest/mock.py:774: AttributeError
```

--- a/tests/test_connection.py
+++ b/tests/test_connection.py
@@ -262,7 +262,7 @@ async def test_start_connection_cannot_increase_recv_buffer(
             raise OSError("Socket error")
 
     mock_socket: socket.socket = create_autospec(
-        socket.socket, spec_set=True, instance=True, name="bad_buffer_socket"
+        socket.socket, spec_set=True, instance=True
     )
     mock_socket.type = socket.SOCK_STREAM
     mock_socket.fileno.return_value = 1
@@ -311,7 +311,7 @@ async def test_start_connection_can_only_increase_buffer_size_to_262144(
                 raise OSError("Socket error")
 
     mock_socket: socket.socket = create_autospec(
-        socket.socket, spec_set=True, instance=True, name="bad_buffer_socket"
+        socket.socket, spec_set=True, instance=True
     )
     mock_socket.type = socket.SOCK_STREAM
     mock_socket.fileno.return_value = 1

